Parsers/ASimWebSession/Parsers/ASimWebSessionFortinetFortiGate.yaml

```diff
@@ -0,0 +1,138 @@
+- Title: Fortinet FortiGate syslogd documentation
+  Link: https://docs.fortinet.com/document/fortigate/7.2.0/cli-reference/450620/config-log-syslogd-setting
+- Title: Fortinet FortiGate extended logging for user agent, http method and x-forwarded-for
+  Link: https://docs.fortinet.com/document/fortigate/7.4.0/fortios-log-message-reference/496081/enabling-extended-logging
+Description: |
+  This ASIM parser supports filtering and normalizing Fortinet FortiGate logs produced by the Microsoft Sentinel Fortinet connector to the ASIM Web Session normalized schema.
+ParserName: ASimWebSessionFortinetFortiGate
+EquivalentBuiltInParser: _ASim_WebSession_FortinetFortiGate
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let EventLookup=datatable(DeviceAction:string,DvcAction:string,EventResult:string)
+  [
+    "passthrough","Allow","Success"
+    , "blocked","Deny","Failure"
+  ];
+  // -- See https://docs.fortinet.com/document/fortigate/7.2.4/fortios-log-message-reference/671442/cef-priority-levels
+  let SeverityLookup = datatable (EventOriginalSeverity:string, EventSeverity:string)
+  [
+    "1", "Informational", // Debug
+    "2", "Informational", // Information
+    "3", "Informational", // Notification
+    "4", "Low", // Warning
+    "5", "Low", // Error
+    "6", "High", // Critical
+    "7", "Medium", // Alert
+    "8", "High" // Emergency
+  ];
+  let parser=(disabled:bool=false){
+    CommonSecurityLog
+    | where not(disabled)
+    | where DeviceVendor == "Fortinet" 
+        and DeviceProduct startswith "FortiGate"
+        and (Activity == "webfilter utm passthrough"
+            or Activity == "webfilter utm blocked")
+    | extend EventResultDetails = "NA"
+    | lookup EventLookup on DeviceAction 
+    | project Activity,AdditionalExtensions,DestinationIP,DestinationPort,DeviceAction,DeviceInboundInterface,DeviceOutboundInterface,DeviceProduct,DeviceVersion,LogSeverity,Protocol,ReceivedBytes,SentBytes,SourceIP,SourcePort,TimeGenerated, DeviceExternalID, Type, _ItemId, Computer, EventResult, EventResultDetails, DvcAction, RequestURL, RequestContext
+    | project-rename 
+      Url = RequestURL
+      ,UrlCategory = RequestContext
+      ,DstBytes = ReceivedBytes
+      , DstInterfaceName = DeviceOutboundInterface
+      , DstIpAddr = DestinationIP
+      , DstPortNumber = DestinationPort
+      , Dvc = Computer
+      , EventMessage = Activity
+      , EventOriginalSeverity = LogSeverity
+      , EventProduct = DeviceProduct
+      , EventProductVersion = DeviceVersion
+      , SrcBytes = SentBytes
+      , SrcInterfaceName = DeviceInboundInterface
+      , SrcIpAddr = SourceIP
+      , SrcPortNumber = SourcePort
+      , DvcId = DeviceExternalID
+      , EventUid = _ItemId
+    | invoke _ASIM_ResolveNetworkProtocol ('Protocol')
+    | project-rename DvcOriginalAction = DeviceAction
+    | parse-kv AdditionalExtensions as (
+            httpmethod:string,
+            agent:string
```

Comment:
Found different logs. Have shared an email with you for updating logic of these two fields. Please have a look.

Reply:
Switched logic